Jenkinsfile.nightly-build

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import java.text.SimpleDateFormat
+
+properties([pipelineTriggers([cron('15 2 * * *')])])
+
+def dateFormat = new SimpleDateFormat("yyyyMMdd")
+def date = new Date()
+def timestamp = dateFormat.format(date)
+
+pipeline {
+    agent { 
+        node { 
+            label 'nightly-build'
+            // Forcing a short path workspace to prevent issues with arguments lentgh
+            customWorkspace "workspace/${URLDecoder.decode(JOB_NAME)}/${BUILD_NUMBER}"
+        }
+    }
+    stages {
+        stage('Generate release') {
+            steps {
+                sh """
+                npm install
+                node ./node_modules/.bin/grunt release
+                jq -r '.version' config.json > version
+                """
+                script {
+                    env.LISK_VERSION = readFile("version").trim()
+                }
+            }
+        }
+        stage('Prepare build environment') {
+            steps {
+                dir('lisk-build') {
+                    git url: "https://github.com/LiskHQ/lisk-build.git", branch: "1.0.0"
+                }
+                sh "cp release/lisk-${env.LISK_VERSION}.tgz lisk-build/src"
```

Comment:
This code blocks assumes that you are inside `lisk` core directory so you can copy release from a relative path `release/` but if that's true then cloning `lisk-build` inside core repo is not a good idea. Clone it outside main repo.